rules/manager.go

```diff
@@ -901,6 +905,12 @@ func (m *Manager) Update(interval time.Duration, files []string, externalLabels
 	// Stop remaining old groups.
 	for n, oldg := range m.groups {
 		oldg.stop()
+		for _, rule := range oldg.seriesInPreviousEval {
+			for _, s := range rule {
+				oldg.staleSeries = append(oldg.staleSeries, s)
+			}
+		}
+		oldg.cleanupStaleSeries(time.Now())
```

Comment:
You're doing this too early. What if a new rules is created with the same name before when the next eval interval will be?

Reply:
Done. It will make reloads longer but now is right.

Reply:
Note that to be completely safe we probably should wait 2*time.interval because the rule groups are delayed by a seed.